models/demos/falcon7b/tt/falcon_attention.py

```diff
@@ -271,8 +316,14 @@ def forward(
                     key_layer[i],
                     -2,
                     -1,
-                    output_mem_config=self.model_config["K_TRANSPOSED_OUTPUT_MEMCFG"],
-                )
+                    output_mem_config=(
+                        self.model_config["K_TRANSPOSED_OUTPUT_MEMCFG"]
+                        if llm_mode == "prefill" or self.model_config["l1_sharded"] == False
+                        else tt_lib.tensor.MemoryConfig(
+                            tt_lib.tensor.TensorMemoryLayout.HEIGHT_SHARDED, tt_lib.tensor.BufferType.L1
+                        )
+                    ),
+                ),
```

Comment:
It might be better to have this memory config set in model_config.py

Reply:
we do not have separate model_config for prefill and decode, hence, I have to select this here